pkg/search/result.go

```diff
@@ -104,6 +169,63 @@ func (repo Repository) ExportData(fields []string) map[string]interface{} {
 				"id":    repo.Owner.ID,
 				"login": repo.Owner.Login,
 				"type":  repo.Owner.Type,
+				"url":   repo.Owner.URL,
+			}
+		default:
+			sf := fieldByName(v, f)
+			data[f] = sf.Interface()
+		}
+	}
+	return data
+}
+
+func (issue Issue) IsPullRequest() bool {
+	return issue.PullRequestLinks.URL != ""
+}
+
+func (issue Issue) ExportData(fields []string) map[string]interface{} {
+	v := reflect.ValueOf(issue)
+	data := map[string]interface{}{}
+	for _, f := range fields {
+		switch f {
+		case "assignees":
+			assignees := make([]interface{}, 0, len(issue.Assignees))
+			for _, assignee := range issue.Assignees {
+				assignees = append(assignees, map[string]interface{}{
+					"id":    assignee.ID,
+					"login": assignee.Login,
+					"type":  assignee.Type,
+					"url":   assignee.URL,
+				})
+			}
+			data[f] = assignees
+		case "author":
+			data[f] = map[string]interface{}{
+				"id":    issue.Author.ID,
+				"login": issue.Author.Login,
+				"type":  issue.Author.Type,
+				"url":   issue.Author.URL,
+			}
+		case "isPullRequest":
+			data[f] = issue.IsPullRequest()
+		case "labels":
+			labels := make([]interface{}, 0, len(issue.Labels))
+			for _, label := range issue.Labels {
+				labels = append(labels, map[string]interface{}{
+					"color":       label.Color,
+					"description": label.Description,
+					"id":          label.ID,
+					"name":        label.Name,
+					"url":         label.URL,
+				})
+			}
+			data[f] = labels
+		case "repository":
+			comp := strings.Split(issue.RepositoryURL, "/")
+			name := strings.Join(comp[len(comp)-2:], "/")
+			data[f] = map[string]interface{}{
+				"name": name,
+				"url":  issue.RepositoryURL,
```

Comment:
For the `url` property in other commands we output the browser URL, not the API URL. But, I don't think it's necessary to include this property here. For identifying the repository, just `nameWithOwner` will do, and if the clients needs to construct the browser URL for this repo, it's trivial to do so.

Optionally we can consider also adding the repository's `id` field instead of `url`. That would be useful for scripting.

Reply:
Agreed. The API doesn't return the `id` fields so going to leave it as just `nameWithOwner` for now.